addons/sale_loyalty/models/sale_order.py

```diff
@@ -361,12 +361,12 @@ def _get_reward_values_discount(self, reward, coupon, **kwargs):
             }]
         discount_factor = min(1, (max_discount / discountable)) if discountable else 1
         mapped_taxes = {tax: self.fiscal_position_id.map_tax(tax) for tax in discountable_per_tax}
+        name_translated = _('Discount: %(desc)s%(tax_str)s')
         reward_dict = {tax: {
-            'name': _(
-                'Discount: %(desc)s%(tax_str)s',
-                desc=reward.description,
-                tax_str=len(discountable_per_tax) and any(t.name for t in mapped_taxes[tax]) and _(' - On product with the following taxes: %(taxes)s', taxes=", ".join(mapped_taxes[tax].mapped('name'))) or '',
-            ),
+            'name': name_translated % {
+                'desc' : reward.description,
+                'tax_str' : len(discountable_per_tax) and any(t.name for t in mapped_taxes[tax]) and _(' - On product with the following taxes: %(taxes)s', taxes=", ".join(mapped_taxes[tax].mapped('name'))) or '',
```

Comment:
Are we sure the problem doesn't happen for `_(' - On product with the following taxes: %(taxes)s'` as well?

Reply:
Good catch, It was indeed not translated ! This has been fixed in the same way ;)